src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py

```diff
@@ -441,6 +445,26 @@ def ensure_container_insights_for_monitoring(
             # get existing tags on the container insights extension DCR if the customer added any
             existing_tags = get_existing_container_insights_extension_dcr_tags(
                 cmd, dcr_url)
+            # get data collection settings
+            extensionSettings = {}
+            if data_collection_settings is not None:
+                dataCollectionSettings = _get_data_collection_settings(data_collection_settings)
+                if 'interval' in dataCollectionSettings.keys():
+                    intervalValue = dataCollectionSettings["interval"]
+                if (bool(re.match(r'^[0-9]+[m]$', intervalValue))) is False:
+                    raise InvalidArgumentValueError('interval format must be in <number>m')
+                intervalValue = int(intervalValue.rstrip("m"))
+                if intervalValue <= 0 or intervalValue > 30:
+                    raise InvalidArgumentValueError('interval value MUST be in the range from 1m to 30m')
+                if 'namespaceFilteringMode' in dataCollectionSettings.keys():
+                    namespaceFilteringModeValue = dataCollectionSettings["namespaceFilteringMode"].lower()
+                    if namespaceFilteringModeValue not in ["off", "exclude", "include"]:
+                        raise InvalidArgumentValueError('namespaceFilteringMode value MUST be either Off or Exclude or Include')
+                if 'namespaces' in dataCollectionSettings.keys():
+                    namspaces = dataCollectionSettings["namespaces"]
+                    if isinstance(namspaces, list) is False:
+                        raise InvalidArgumentValueError('namespaces must be an array type')
```

Comment:
I suppose it would be better to put these validations to some other function and add some unit tests.

Reply:
> I suppose it would be better to put these validations to some other function and add some unit tests.

> fixed.